src/Adapter/Cart/AbstractCartHandler.php

```diff
@@ -61,4 +69,38 @@ protected function getCart(CartId $cartId)
 
         return $cart;
     }
+
+    /**
+     * @param ContextStateManager $contextStateManager
+     * @param Cart $cart
+     */
+    protected function setCartContext(ContextStateManager $contextStateManager, Cart $cart): void
+    {
+        $contextStateManager
+            ->saveCurrentContext()
+            ->setCart($cart)
+            ->setCustomer(new Customer($cart->id_customer))
+            ->setCurrency(new Currency($cart->id_currency))
+            ->setLanguage(new Language($cart->id_lang))
+            ->setCountry($this->getCartTaxCountry($cart))
+            ->setShop(new Shop($cart->id_shop))
+        ;
+    }
+
+    /**
+     * @param Cart $cart
+     *
+     * @return Country
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    protected function getCartTaxCountry(Cart $cart): Country
+    {
+        $taxAddressType = Configuration::get('PS_TAX_ADDRESS_TYPE');
+        $taxAddressId = property_exists($cart, $taxAddressType) ? $cart->{$taxAddressType} : $cart->id_address_delivery;
```

Comment:
Watch out with `property_exists` it doesn't check if the property is accessible.
```suggestion
        $taxAddressId = $cart->{$taxAddressType} ?? $cart->id_address_delivery;
```